### PR TITLE
pitr:  Compatible with tiflash

### LINKED ISSUE #34777: pitr: Compatible with tiflash
## Enhancement
- Support Tiflash replica on `br restore full`(restore snapshot)
- Do not support Tiflash replica on `br restore point`(PiTR)

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?

Issue Number: close #34777

Problem Summary:

### What is changed and how it works?
- update global schema version( inc 128) after restore log,  thus the `tiflash` will skip the `schemaDiff`.
- Support Tiflash replica on `br restore full`, bug do not support Tiflash replica on `pitr`.
- Remove scheduler when restore, and recover scheduler at the end.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

#### Manual test
1. The tiflash replica is restored to down-stream if use `br restore full`.(restore full)
2. The tiflash replica isn't restored to down-stream if use `br restore point`. ( PiTR)

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by 3pointer — COMMENTED** [review]
rest LGTM

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -1004,7 +1005,18 @@ func restoreStream(
 		ctx = opentracing.ContextWithSpan(ctx, span1)
 	}
 
-	client, err := createRestoreClient(ctx, g, cfg)
+	mgr, err := NewMgr(ctx, g, cfg.PD, cfg.TLS, GetKeepalive(&cfg.Config),
+		cfg.CheckRequirements, true)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() {
```
Comment (3pointer) [inline br/pkg/task/stream.go:1013]: maybe just simple use defer mgr.close() is enough.

**Review by 3pointer — APPROVED** [review]
rest LGTM

**Review by fengou1 — APPROVED** [review]

**Pull request merged**